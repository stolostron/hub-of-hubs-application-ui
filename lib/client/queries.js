/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project

import gql from 'graphql-tag'

export const QueryApplicationList = gql`
  query applicationsList {
    applications {
      _uid
      apiVersion
      name
      namespace
      cluster
      dashboard
      clusterCount
      hubChannels
      hubSubscriptions {
        _uid
        name
        localPlacement
        timeWindow
        status
        channel
      }
      created
      applicationSet
      destinationCluster
      destinationNamespace
      repoURL
      path
      chart
      targetRevision
    }
  }
`

export const QuerySubscriptionList = gql`
  query subscriptionsList {
    subscriptions {
      _uid
      name
      namespace
      appCount
      clusterCount
      localPlacement
      timeWindow
      status
      channel
      created
    }
  }
`

export const QueryPlacementRuleList = gql`
  query placementRulesList {
    placementRules {
      _uid
      name
      namespace
      clusterCount
      replicas
      created
    }
  }
`

export const QueryChannelList = gql`
  query channelsList {
    channels {
      _uid
      name
      namespace
      subscriptionCount
      clusterCount
      type
      pathname
      localPlacement
      created
    }
  }
`

export const HCMChannel = gql`
  query getSingleChannel($name: String!, $namespace: String!) {
    items: channels(name: $name, namespace: $namespace) {
      metadata {
        labels
        name
        namespace
        uid
        selfLink
        resourceVersion
        creationTimestamp
      }
      type
      secretRef
      objectPath
      raw
    }
  }
`

export const HCMSubscription = gql`
  query getSingleSubscription($name: String!, $namespace: String!) {
    items: subscriptions(name: $name, namespace: $namespace) {
      metadata {
        labels
        name
        namespace
        uid
        selfLink
        resourceVersion
        creationTimestamp
      }
      raw
    }
  }
`

export const HCMPlacementRule = gql`
  query getSinglePlacementRule($name: String!, $namespace: String!) {
    items: placementrules(name: $name, namespace: $namespace) {
      metadata {
        labels
        name
        namespace
        uid
        selfLink
        resourceVersion
        creationTimestamp
      }
      raw
    }
  }
`

export const HCMChannelList = gql`
  query getChannels {
    items: channels {
      metadata {
        labels
        name
        namespace
        uid
        selfLink
        resourceVersion
        creationTimestamp
      }
      type
      secretRef
      objectPath
      raw
    }
  }
`

export const gitChannelBranches = gql`
  query getGitChannelBranches(
    $gitUrl: String!
    $namespace: String
    $secretRef: String
    $user: String
    $accessToken: String
  ) {
    items: gitChannelBranches(
      gitUrl: $gitUrl
      namespace: $namespace
      secretRef: $secretRef
      user: $user
      accessToken: $accessToken
    )
  }
`

export const gitChannelPaths = gql`
  query getGitChannelPaths(
    $gitUrl: String!
    $branch: String!
    $path: String
    $namespace: String
    $secretRef: String
    $user: String
    $accessToken: String
  ) {
    items: gitChannelPaths(
      gitUrl: $gitUrl
      branch: $branch
      path: $path
      namespace: $namespace
      secretRef: $secretRef
      user: $user
      accessToken: $accessToken
    )
  }
`
export const argoAppRouteURL = gql`
  query getArgoAppRouteURL(
    $cluster: String!
    $namespace: String!
    $name: String!
    $apiVersion: String!
  ) {
    argoAppRouteURL(
      cluster: $cluster
      namespace: $namespace
      name: $name
      apiVersion: $apiVersion
    )
  }
`

export const argoServerNS = gql`
  query getargoServerNS {
    argoServers {
      argoServerNS
    }
  }
`

export const HCMAnsibleTower = gql`
  query getSecrets($label: String!, $value: String!) {
    secrets(label: $label, value: $value) {
      ansibleSecretName
      ansibleSecretNamespace
    }
  }
`

export const managedClusterStatus = gql`
  query getManagedCluster($clusterName: String!) {
    isManagedClusterConditionAvailable(clusterName: $clusterName)
  }
`

export const routeResourceURL = gql`
  query getRouteResourceURL(
    $cluster: String!
    $namespace: String!
    $name: String!
    $apiVersion: String!
  ) {
    routeResourceURL(
      cluster: $cluster
      namespace: $namespace
      name: $name
      apiVersion: $apiVersion
    )
  }
`

export const HCMContainer = gql`
  query getResource(
    $selfLink: String
    $namespace: String
    $kind: String
    $name: String
    $cluster: String
  ) {
    getResource(
      selfLink: $selfLink
      namespace: $namespace
      kind: $kind
      name: $name
      cluster: $cluster
    )
  }
`

export const HCMSubscriptionList = gql`
  query searchSchema {
    subscriptions {
      metadata {
        name
        namespace
        creationTimestamp
      }
      raw
    }
  }
`

export const HCMPlacementRuleList = gql`
  query getPlacementRules($namespace: String) {
    placementrules(namespace: $namespace) {
      metadata {
        name
        namespace
        creationTimestamp
      }
      raw
    }
  }
`

export const HCMPlacementList = gql`
  query getPlacements($namespace: String) {
    placements(namespace: $namespace) {
      metadata {
        name
        namespace
        creationTimestamp
      }
      raw
    }
  }
`

export const HCMApplicationList = gql`
  query searchResult($input: [SearchInput]) {
    searchResult: search(input: $input) {
      items
    }
  }
`

export const getApplication = gql`
  query getApplication($name: String!, $namespace: String!) {
    application(name: $name, namespace: $namespace) {
      metadata {
        annotations
        creationTimestamp
        labels
        name
        namespace
        resourceVersion
        selfLink
        uid
      }
      name
      namespace
      app
      subscriptions
    }
  }
`

export const getArgoApplication = gql`
  query getApplication(
    $name: String!
    $namespace: String!
    $apiversion: String
  ) {
    application(name: $name, namespace: $namespace, apiversion: $apiversion) {
      metadata {
        annotations
        creationTimestamp
        labels
        name
        namespace
        resourceVersion
        selfLink
        uid
      }
      name
      namespace
      app
      subscriptions
      placement
    }
  }
`

export const getApplications = gql`
  query getApplications {
    applications {
      metadata {
        annotations
        creationTimestamp
        labels
        name
        namespace
        resourceVersion
        uid
      }
      raw
    }
  }
`

export const getApplicationSetRelatedResources = gql`
  query applicationSetRelatedResources($name: String, $namespace: String) {
    applicationSetRelatedResources(name: $name, namespace: $namespace)
  }
`

export const createApplication = gql`
  mutation createApplication($application: [JSON]!) {
    createApplication(application: $application)
  }
`

export const updateApplication = gql`
  mutation updateApplication($application: [JSON]!) {
    updateApplication(application: $application)
  }
`

export const createResources = gql`
  mutation createResources($resources: [JSON]!) {
    createResources(resources: $resources)
  }
`

export const removeQuery = gql`
  mutation deleteQuery($resource: JSON!) {
    deleteQuery(resource: $resource)
  }
`

export const updateResource = gql`
  mutation updateResource(
    $resourceType: String!
    $namespace: String!
    $name: String!
    $body: JSON
    $selfLink: String
    $resourcePath: String
  ) {
    updateResource(
      resourceType: $resourceType
      namespace: $namespace
      name: $name
      body: $body
      selfLink: $selfLink
      resourcePath: $resourcePath
    )
  }
`

export const deleteResource = gql`
  mutation deleteResource(
    $apiVersion: String
    $kind: String
    $name: String
    $namespace: String
    $childResources: JSON
  ) {
    deleteResource(
      apiVersion: $apiVersion
      kind: $kind
      name: $name
      namespace: $namespace
      childResources: $childResources
    )
  }
`

export const HCMPod = gql`
  query getPod($name: String!, $namespace: String!, $clusterName: String!) {
    items: pod(name: $name, namespace: $namespace, clusterName: $clusterName) {
      cluster {
        metadata {
          name
        }
        clusterip
      }
      hostIP
      images
      containers {
        name
      }
      metadata {
        creationTimestamp
        labels
        name
        namespace
        selfLink
      }
      podIP
      restarts
      status
    }
  }
`

export const HCMTopologyFilters = gql`
  query getTopologyFilters {
    clusters {
      metadata {
        name
        labels
      }
    }
    namespaces {
      metadata {
        name
      }
    }
    labels {
      name
      value
    }
    resourceTypes
  }
`

export const HCMTopology = gql`
  query getTopology($filter: TopologyFilter) {
    topology(filter: $filter) {
      resources {
        id
        uid
        name
        cluster
        clusterName
        type
        specs
        namespace
        topology
        labels {
          name
          value
        }
      }
      relationships {
        type
        specs
        to {
          uid
        }
        from {
          uid
        }
      }
      error
    }
  }
`

export const createUserQueries = gql`
  mutation saveQuery($resource: JSON!) {
    saveQuery(resource: $resource)
  }
`

export const userAccess = gql`
  query userAccess(
    $resource: String
    $kind: String
    $action: String!
    $namespace: String
    $apiGroup: String
    $version: String
  ) {
    userAccess(
      resource: $resource
      kind: $kind
      action: $action
      namespace: $namespace
      apiGroup: $apiGroup
      version: $version
    )
  }
`

export const userAccessAnyNamespaces = gql`
  query userAccessAnyNamespaces(
    $resource: String!
    $action: String!
    $apiGroup: String
  ) {
    userAccessAnyNamespaces(
      resource: $resource
      action: $action
      apiGroup: $apiGroup
    )
  }
`

export const userInfo = gql`
  query userInfo {
    items: userInfo {
      userId
      activeAccountId
      activeAccountName
    }
  }
`
export const HCMNamespaceList = gql`
  query applicationNamespaces {
    items: applicationNamespaces {
      metadata {
        annotations
        name
      }
    }
  }
`

export const HCMNamespace = gql`
  query applicationNamespaces($namespace: String!) {
    items: applicationNamespaces(namespace: $namespace) {
      metadata {
        annotations
        name
      }
    }
  }
`
