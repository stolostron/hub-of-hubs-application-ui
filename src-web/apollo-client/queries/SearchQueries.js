/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import gql from 'graphql-tag'

export const GET_RESOURCE = gql`
  query getResource(
    $selfLink: String
    $namespace: String
    $kind: String
    $name: String
    $cluster: String
  ) {
    getResource(
      selfLink: $selfLink
      namespace: $namespace
      kind: $kind
      name: $name
      cluster: $cluster
    )
  }
`

export const GET_SEARCH_SCHEMA = gql`
  query searchSchema {
    searchSchema
  }
`

export const GET_SEARCH_COMPLETE = gql`
  query searchComplete($property: String!, $query: SearchInput) {
    searchComplete(property: $property, query: $query)
  }
`

export const GET_SEARCH_INPUT_TEXT = gql`
  {
    searchInput @client {
      text
    }
  }
`

export const GET_RELATED_RESOURCES = gql`
  {
    relatedResources @client {
      visibleKinds
    }
  }
`

export const SEARCH_QUERY = gql`
  query searchResult($input: [SearchInput]) {
    searchResult: search(input: $input) {
      items
    }
  }
`

export const SEARCH_QUERY_RELATED = gql`
  query searchResult($input: [SearchInput]) {
    searchResult: search(input: $input) {
      items
      related {
        kind
        items
      }
    }
  }
`

export const GET_SAVED_USER_QUERY = gql`
  query userQueries {
    items: userQueries {
      name
      description
      searchText
    }
  }
`

export const SAVE_USER_QUERY = gql`
  mutation saveQuery($resource: JSON!) {
    saveQuery(resource: $resource)
  }
`
