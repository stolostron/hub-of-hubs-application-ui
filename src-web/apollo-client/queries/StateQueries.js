/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import gql from 'graphql-tag'

export const GET_MODAL_STATE = gql`
  {
    modal @client {
      open
      type
      data {
        name
        description
        searchText
      }
    }
  }
`

export const GET_ACTION_MODAL_STATE = gql`
  {
    actionModal @client {
      open
      type
      resourceType {
        name
        list
      }
      data {
        apiVersion
        name
        namespace
        clusterName
        selfLink
        _uid
        kind
        itemGroup {
          destinationNamespace
          name
          _uid
        }
      }
    }
  }
`

export const UPDATE_MODAL = gql`
  mutation UpdateModal(
    $__typename: String
    $open: Boolean
    $type: String
    $data: JSON
  ) {
    updateModal(__typename: $__typename, open: $open, type: $type, data: $data)
      @client
  }
`

export const UPDATE_ACTION_MODAL = gql`
  mutation UpdateActionModal(
    $__typename: String
    $open: Boolean
    $type: String
    $resourceType: JSON
    $data: JSON
  ) {
    updateActionModal(
      __typename: $__typename
      open: $open
      type: $type
      resourceType: $resourceType
      data: $data
    ) @client
  }
`
