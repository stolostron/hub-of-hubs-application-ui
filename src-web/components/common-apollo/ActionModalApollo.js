/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import _ from 'lodash'
import React from 'react'
import PropTypes from 'prop-types'
import loadable from '@loadable/component'
import { GET_ACTION_MODAL_STATE } from '../../apollo-client/queries/StateQueries'
import { Query } from 'react-apollo'

let RemoveResourceModal
let SyncResourceModal

class ActionModalApollo extends React.PureComponent {
  getMatchedModal = ({ type, resourceType, open, data }) => {
    switch (type) {
    case 'table.actions.applications.remove':
    case 'table.actions.subscriptions.remove':
    case 'table.actions.placementrules.remove':
    case 'table.actions.channels.remove':
    case 'table.actions.remove': {
      return (
        open &&
          this.getRemoveResourceModal({
            open: true,
            type: 'resource-remove',
            resourceType,
            label: {
              primaryBtn: 'modal.remove.submit',
              label: `modal.remove-${resourceType.name.toLowerCase()}.label`,
              heading: `modal.remove-${resourceType.name.toLowerCase()}.heading`
            },
            data: data
          })
      )
    }

    case 'actions.sync.application': {
      return (
        open &&
          this.getSyncResourceModal({
            open: true,
            type: 'resource-sync',
            resourceType,
            label: {
              primaryBtn: 'modal.sync.submit',
              label: `modal.sync-${resourceType.name.toLowerCase()}.label`,
              heading: `modal.sync-${resourceType.name.toLowerCase()}.heading`
            },
            data: data
          })
      )
    }

    default:
      return null
    }
  };

  getRemoveResourceModal = props => {
    RemoveResourceModal =
      RemoveResourceModal === undefined
        ? loadable(() =>
            import(/* webpackChunkName: "remove-resource-modal" */ '../modals/RemoveResourceModal')
        )
        : RemoveResourceModal
    return this.getModal(RemoveResourceModal, props)
  };

  getSyncResourceModal = props => {
    SyncResourceModal =
      SyncResourceModal === undefined
        ? loadable(() =>
            import(/* webpackChunkName: "sync-resource-modal" */ '../modals/SyncResourceModal')
        )
        : SyncResourceModal
    return this.getModal(SyncResourceModal, props)
  };

  getModal = (Component, props) => <Component {...props} />;

  render() {
    return (
      <Query query={GET_ACTION_MODAL_STATE}>
        {({ data }) => {
          if (_.get(data, 'actionModal.open')) {
            return this.getMatchedModal(_.get(data, 'actionModal'))
          } else {
            return null
          }
        }}
      </Query>
    )
  }
}

ActionModalApollo.contextTypes = {
  locale: PropTypes.string
}

ActionModalApollo.propTypes = {
  locale: PropTypes.string
}

export default ActionModalApollo
