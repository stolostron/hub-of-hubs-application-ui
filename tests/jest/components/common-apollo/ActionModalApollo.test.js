/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
/* eslint-disable no-unused-vars */

import React from "react";
import { MockedProvider } from "react-apollo/test-utils";
import renderer from "react-test-renderer";
import ActionModalApollo from "../../../../src-web/components/common-apollo/ActionModalApollo";
import { GET_ACTION_MODAL_STATE } from "../../../../src-web/apollo-client/queries/StateQueries";

const delay = ms =>
  new Promise(resolve => {
    setTimeout(() => {
      resolve();
    }, ms);
  });

const mocks = {
  invalidMock: {
    request: { query: GET_ACTION_MODAL_STATE },
    result: {
      data: {
        actionModal: {
          open: true,
          type: "invalid",
          __typename: "actionModal",
          resourceType: {
            _uid: "invalid",
            name: "invalid",
            list: "invalid"
          },
          data: {
            _uid: "invalid",
            name: "invalid",
            namespace: "invalid",
            clusterName: "invalid",
            selfLink: "invalid",
            kind: "invalid"
          }
        }
      }
    }
  },
  editMock: {
    request: { query: GET_ACTION_MODAL_STATE },
    result: {
      data: {
        actionModal: {
          open: true,
          type: "table.actions.edit",
          __typename: "actionModal",
          resourceType: {
            name: "HCMPod",
            list: "HCMPodList"
          },
          data: {
            _uid: "icp-mongodb-0",
            name: "icp-mongodb-0",
            namespace: "kube-system",
            clusterName: "local-cluster",
            selfLink: "/api/v1/namespaces/kube-system/pods/icp-mongodb-0",
            kind: "pods"
          }
        }
      }
    }
  },
  editAppMock: {
    request: { query: GET_ACTION_MODAL_STATE },
    result: {
      data: {
        actionModal: {
          open: true,
          type: "table.actions.applications.edit",
          __typename: "actionModal",
          resourceType: {
            name: "HCMApplication",
            list: "HCMApplicationList"
          },
          data: {
            _uid: "icp-mongodb-0",
            name: "icp-mongodb-0",
            namespace: "kube-system",
            clusterName: "local-cluster",
            selfLink: "/api/v1/namespaces/kube-system/pods/icp-mongodb-0",
            kind: "applications"
          }
        }
      }
    }
  },
  removeMock: {
    request: { query: GET_ACTION_MODAL_STATE },
    result: {
      data: {
        actionModal: {
          open: true,
          type: "table.actions.remove",
          __typename: "actionModal",
          resourceType: {
            name: "HCMApplication",
            list: "HCMApplicationList"
          },
          data: {
            _uid: "icp-mongodb-0",
            name: "icp-mongodb-0",
            namespace: "kube-system",
            clusterName: "local-cluster",
            selfLink: "/api/v1/namespaces/kube-system/pods/icp-mongodb-0",
            kind: "applications"
          }
        }
      }
    }
  },
  removeAppMock: {
    request: { query: GET_ACTION_MODAL_STATE },
    result: {
      data: {
        actionModal: {
          open: true,
          type: "table.actions.applications.remove",
          __typename: "actionModal",
          resourceType: {
            name: "HCMApplication",
            list: "HCMApplicationList"
          },
          data: {
            _uid: "icp-mongodb-0",
            name: "icp-mongodb-0",
            namespace: "kube-system",
            clusterName: "local-cluster",
            selfLink: "/api/v1/namespaces/kube-system/pods/icp-mongodb-0",
            kind: "applications"
          }
        }
      }
    }
  }
};

window.open = () => {}; // provide an empty implementation for window.open
describe("ActionModalApollo Testing", () => {
  it("To Return Null For Invalid Table Action", async () => {
    const component = renderer.create(
      <MockedProvider mocks={[mocks.invalidMock]} addTypename={false}>
        <ActionModalApollo locale={"en-US"} />
      </MockedProvider>
    );
    await delay(0);
    expect(component.toJSON()).toEqual(null);
  });

  it("Changes Apollo Client Cache For Edit Modal", async () => {
    const component = renderer.create(
      <MockedProvider mocks={[mocks.editMock]} addTypename={false}>
        <ActionModalApollo locale={"en-US"} />
      </MockedProvider>
    );
    await delay(0);
    expect(
      component.getInstance().state.client.cache.data.data
    ).toMatchSnapshot();
  });

  it("Changes Apollo Client Cache For Edit App Modal", async () => {
    const component = renderer.create(
      <MockedProvider mocks={[mocks.editAppMock]} addTypename={false}>
        <ActionModalApollo locale={"en-US"} />
      </MockedProvider>
    );
    await delay(0);
    expect(
      component.getInstance().state.client.cache.data.data
    ).toMatchSnapshot();
  });

  it("Changes Apollo Client Cache For Remove Resource Modal", async () => {
    const component = renderer.create(
      <MockedProvider mocks={[mocks.removeMock]} addTypename={false}>
        <ActionModalApollo locale={"en-US"} />
      </MockedProvider>
    );
    await delay(0);
    expect(
      component.getInstance().state.client.cache.data.data
    ).toMatchSnapshot();
  });

  it("Changes Apollo Client Cache For Remove App Resource Modal", async () => {
    const component = renderer.create(
      <MockedProvider mocks={[mocks.removeAppMock]} addTypename={false}>
        <ActionModalApollo locale={"en-US"} />
      </MockedProvider>
    );
    await delay(0);
    expect(
      component.getInstance().state.client.cache.data.data
    ).toMatchSnapshot();
  });
});
