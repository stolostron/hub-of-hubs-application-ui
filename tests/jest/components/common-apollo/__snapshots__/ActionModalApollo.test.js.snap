// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionModalApollo Testing Changes Apollo Client Cache For Edit App Modal 1`] = `
Object {
  "$ROOT_QUERY.actionModal": Object {
    "data": Object {
      "generated": true,
      "id": "$ROOT_QUERY.actionModal.data",
      "type": "id",
      "typename": undefined,
    },
    "open": true,
    "resourceType": Object {
      "generated": true,
      "id": "$ROOT_QUERY.actionModal.resourceType",
      "type": "id",
      "typename": undefined,
    },
    "type": "table.actions.applications.edit",
  },
  "$ROOT_QUERY.actionModal.data": Object {
    "_uid": "icp-mongodb-0",
    "clusterName": "local-cluster",
    "kind": "applications",
    "name": "icp-mongodb-0",
    "namespace": "kube-system",
    "selfLink": "/api/v1/namespaces/kube-system/pods/icp-mongodb-0",
  },
  "$ROOT_QUERY.actionModal.resourceType": Object {
    "list": "HCMApplicationList",
    "name": "HCMApplication",
  },
  "ROOT_QUERY": Object {
    "actionModal": Object {
      "generated": true,
      "id": "$ROOT_QUERY.actionModal",
      "type": "id",
      "typename": "actionModal",
    },
  },
}
`;

exports[`ActionModalApollo Testing Changes Apollo Client Cache For Edit Modal 1`] = `
Object {
  "$ROOT_QUERY.actionModal": Object {
    "data": Object {
      "generated": true,
      "id": "$ROOT_QUERY.actionModal.data",
      "type": "id",
      "typename": undefined,
    },
    "open": true,
    "resourceType": Object {
      "generated": true,
      "id": "$ROOT_QUERY.actionModal.resourceType",
      "type": "id",
      "typename": undefined,
    },
    "type": "table.actions.edit",
  },
  "$ROOT_QUERY.actionModal.data": Object {
    "_uid": "icp-mongodb-0",
    "clusterName": "local-cluster",
    "kind": "pods",
    "name": "icp-mongodb-0",
    "namespace": "kube-system",
    "selfLink": "/api/v1/namespaces/kube-system/pods/icp-mongodb-0",
  },
  "$ROOT_QUERY.actionModal.resourceType": Object {
    "list": "HCMPodList",
    "name": "HCMPod",
  },
  "ROOT_QUERY": Object {
    "actionModal": Object {
      "generated": true,
      "id": "$ROOT_QUERY.actionModal",
      "type": "id",
      "typename": "actionModal",
    },
  },
}
`;

exports[`ActionModalApollo Testing Changes Apollo Client Cache For Remove App Resource Modal 1`] = `
Object {
  "$ROOT_QUERY.actionModal": Object {
    "data": Object {
      "generated": true,
      "id": "$ROOT_QUERY.actionModal.data",
      "type": "id",
      "typename": undefined,
    },
    "open": true,
    "resourceType": Object {
      "generated": true,
      "id": "$ROOT_QUERY.actionModal.resourceType",
      "type": "id",
      "typename": undefined,
    },
    "type": "table.actions.applications.remove",
  },
  "$ROOT_QUERY.actionModal.data": Object {
    "_uid": "icp-mongodb-0",
    "clusterName": "local-cluster",
    "kind": "applications",
    "name": "icp-mongodb-0",
    "namespace": "kube-system",
    "selfLink": "/api/v1/namespaces/kube-system/pods/icp-mongodb-0",
  },
  "$ROOT_QUERY.actionModal.resourceType": Object {
    "list": "HCMApplicationList",
    "name": "HCMApplication",
  },
  "ROOT_QUERY": Object {
    "actionModal": Object {
      "generated": true,
      "id": "$ROOT_QUERY.actionModal",
      "type": "id",
      "typename": "actionModal",
    },
  },
}
`;

exports[`ActionModalApollo Testing Changes Apollo Client Cache For Remove Resource Modal 1`] = `
Object {
  "$ROOT_QUERY.actionModal": Object {
    "data": Object {
      "generated": true,
      "id": "$ROOT_QUERY.actionModal.data",
      "type": "id",
      "typename": undefined,
    },
    "open": true,
    "resourceType": Object {
      "generated": true,
      "id": "$ROOT_QUERY.actionModal.resourceType",
      "type": "id",
      "typename": undefined,
    },
    "type": "table.actions.remove",
  },
  "$ROOT_QUERY.actionModal.data": Object {
    "_uid": "icp-mongodb-0",
    "clusterName": "local-cluster",
    "kind": "applications",
    "name": "icp-mongodb-0",
    "namespace": "kube-system",
    "selfLink": "/api/v1/namespaces/kube-system/pods/icp-mongodb-0",
  },
  "$ROOT_QUERY.actionModal.resourceType": Object {
    "list": "HCMApplicationList",
    "name": "HCMApplication",
  },
  "ROOT_QUERY": Object {
    "actionModal": Object {
      "generated": true,
      "id": "$ROOT_QUERY.actionModal",
      "type": "id",
      "typename": "actionModal",
    },
  },
}
`;
